zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobTimeoutChecker.java

```diff
@@ -75,12 +85,29 @@ private final class DeactivateTimeOutJobs implements Task {
 
     @Override
     public TaskResult execute(final TaskResultBuilder taskResultBuilder) {
-      final long now = currentTimeMillis();
-      state.forEachTimedOutEntry(
-          now,
-          (key, record) -> taskResultBuilder.appendCommandRecord(key, JobIntent.TIME_OUT, record));
+      if (executionTimestamp == -1) {
+        executionTimestamp = currentTimeMillis();
+      }
+
+      final var counter = new MutableInteger(0);
+      final BooleanSupplier exceedsBatchLimit = () -> counter.getAndIncrement() >= batchLimit;
```

Comment:
@korthout I've moved the batchLimit comparison here and passed a `BooleanSupplier` instead. I have a feeling that there could have been a better solution, but this does the job. WDYT?

CC: @megglos

Reply:
Currently, this does not allow extracting the yielding/restarting-at-index behavior for reuse. The `exceedsBatchLimit` also doesn't fit the abstraction of the JobState, which only provides data access (i.e. it does not know what a batch or a batch limit is).

I suggest only passing the `timestamp`, `index` and `visitor` to the `forEachTimedOutEntry`. This is the bare minimum needed for data access.

The batch limit can be handled inside the visitor, like the message ttl checker. By aligning the implementations already, we can more easily extract the logic for reuse in all checkers.

Reply:
I see what you mean.
The main difference between this and the messageTtlChecker that made it difficult to align both implementation is that this method here `forEachTimedOutEntry` receives a. `callback` method that already returns a `boolean` and that is used in some places. the other one was initially a void and it was easier to manipulate it to fix this.

I'll look into it more by all means